MoreLinq/InternalEnumerableExtensions.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MoreLinq
+{
+    static class InternalEnumerableExtensions
+    {
+        /// <summary>
+        /// Gets the count of an enumerable if it is precalculated in <see cref="ICollection{T}"/>
+        /// or <see cref="IReadOnlyCollection{T}"/>.
+        /// </summary>
+        /// <typeparam name="T">The type of objects to enumerate.</typeparam>
+        /// <param name="source">The enumerable to get the count for.</param>
+        /// <returns>The count if it is available without enumeration, or null if it is not.</returns>
+        public static int? TryGetCollectionCount<T>(this IEnumerable<T> source)
```

Comment:
We don't need to introduce a whole new type just to house this method. There's already `MoreEnumerable` so kindly move this to `MoreEnumerable.cs` and remove the `public` modifier so it becomes private. The effect will be the same but with less ceremony.

Reply:
Right. Forgot everything is in a single class (I need to get out more).